### PR TITLE
UMUS-31 Index Mapping

### PR DESCRIPTION
**Pull request by froboy**
Creates a new "Federated Field" field for aggregating disparate entities into the same field in the solr index.

## To test
- Visit the fields list for your index (like https://labblog.uofmhealth.local/admin/config/search/search-api/index/federated_search_index/fields)
- Click "Add fields" then add a "Federated Field" manage_fields_for_search_index_federated_measured_search_index___university_of_michigan_
- Edit the field label, machine name, and type if you like
- Click "Edit" to modify the field configuration, then add some config. You can either enter plain text or use the token browser to select tokens. manage_fields_for_search_index_federated_measured_search_index___university_of_michigan_
- Notice that if you choose a token that's not available for the specific entity type (e.g. a "user" token on a "node") that you receive an error.
- Save your settings (in the config modal and on the fields page)
- Create/edit and save a piece of content to index it.
- Observe your fields are saved into the solr index using the proper data.

### CODE REVIEW

**Comment by becw** [comment]
Also, would it make sense to add the PR testing instructions to this repo as documentation?

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   ],
   "require": {
     "drupal/core": "^8.0",
-    "drupal/search_api_solr": "^2.0@alpha",
+    "drupal/search_api_solr": "^1.2",
     "drupal/facets": "^1.0@beta"
```
[the remark this answers is not in the document]
Comment (froboy) [reply composer.json]: Facets isn't used for anything (it was going to, but not any more). We should remove that.

File: composer.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
...
   }
```
[the remark this answers is not in the document]
Comment (froboy) [reply composer.json]: Ya def.

[the remark this answers is not in the document]
Comment (froboy) [reply composer.json]: If you can remove this from teh dependencies, I'll remove it from the site config in my next PR.

**Pull request merged**